src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/NeuralNetStreamingExecutor.java

```diff
@@ -0,0 +1,216 @@
+        final VCFHeader vcfHeader = new VCFHeader(hInfo, samples);
+
+        vcfWriter = createVCFWriter(outputFile);
+        vcfWriter.writeHeader(vcfHeader);
+        vcfWriter.close();
+
+        // Start the Python process, and get a FIFO from the executor to use to send data to Python. The lifetime
+        // of the FIFO is managed by the executor; the FIFO will be destroyed when the executor is destroyed.
+        pythonExecutor.start(Collections.emptyList(), true);
+        final File fifoFile = pythonExecutor.getFIFOForWrite();
+
+        // Open the FIFO for writing. Opening a FIFO for read or write will block until there is reader/writer
+        // on the other end, so before we open it, send an ASYNCHRONOUS command (one that doesn't wait for a
+        // response) to the Python process to open the FIFO for reading. The Python process will then block until
+        // we open the FIFO. We can then call getAccumulatedOutput.
+        pythonExecutor.sendAsynchronousCommand(String.format("fifoFile = open('%s', 'r')" + NL, fifoFile.getAbsolutePath()));
+        try {
+            fifoWriter = new FileOutputStream(fifoFile);
+        } catch ( IOException e ) {
+            throw new GATKException("Failure opening FIFO for writing", e);
+        }
+
+        pythonExecutor.getAccumulatedOutput();
+        asyncWriter = pythonExecutor.getAsynchronousStreamWriterService(fifoWriter, AsynchronousStreamWriterService.stringSerializer);
+        batchList = new ArrayList<>(pythonSyncFrequency);
+
+        // Also, ask Python to open our output file, where it will write the contents of everything it reads
+        // from the FIFO. <code sendSynchronousCommand/>
+        pythonExecutor.sendSynchronousCommand(String.format("tempFile = open('%s', 'a')" + NL, outputFile.getAbsolutePath()));
+        pythonExecutor.sendSynchronousCommand("from keras.models import load_model" + NL);
+        pythonExecutor.sendSynchronousCommand("import vqsr_cnn" + NL);
+        pythonExecutor.sendSynchronousCommand(String.format("model = load_model('%s')", architecture) + NL);
+        logger.info("Loaded CNN architecture:"+architecture);
+    }
+
+    @Override
+    public void apply(final VariantContext variant, final ReadsContext readsContext, final ReferenceContext referenceContext, final FeatureContext featureContext) {
+        referenceContext.setWindow(63, 64);
+        transferToPythonViaFifo(variant, referenceContext);
+    }
+
+    private void transferToPythonViaFifo(final VariantContext variant, final ReferenceContext referenceContext) {
+        if (variant.isSNP() || variant.isIndel()) {
+            try {
+                String ref = new String(Arrays.copyOfRange(referenceContext.getBases(), 0, 128), "UTF-8");
+                String varData = getVariantDataString(variant, referenceContext);
+                String isSnp = variant.isSNP() ? "1" : "0";
+                String genos = "\t.";
+                if (noSamples) genos = "";
```

Comment:
These variables all could use less cryptic names; most of them are only used once though so they could be inlined below.

Reply:
inlined